e2e/capacity/capacity_rpc.test.ts

```diff
@@ -0,0 +1,127 @@
+      call.toHex(),
+      null
+    );
+    assert.notEqual(feeDetails, undefined, 'should have returned a feeDetails');
+    assert.notEqual(feeDetails.inclusionFee, undefined, 'should have returned a partialFee');
+    assert(feeDetails.inclusionFee.isSome, 'should have returned a partialFee');
+    const { baseFee, lenFee, adjustedWeightFee } = feeDetails.inclusionFee.toJSON() as any;
+    assert(Math.abs(baseFee - 106382) < 10_000, 'The base fee appears to be wrong or have changed more than expected');
+    assert(Math.abs(lenFee - 1170000) < 100, 'The len fee appears to be wrong or have changed more than expected');
+    // This is comparing stable weight, which has no impact from targeted_fee_adjustment, with actual weights.
+    assert(
+      Math.abs(adjustedWeightFee - weightFee.toNumber()) < 10_000,
+      'The adjusted weight fee appears to be wrong or have changed more than expected'
+    );
+  });
+
+  it('Returns `FeeDetails` when requesting capacity cost of a transaction when wrapped in payWithCapacity', async function () {
+    const addProviderPayload = await generateDelegationPayload({ ...defaultPayload });
+    const addProviderData = ExtrinsicHelper.api.registry.createType('PalletMsaAddProvider', addProviderPayload);
+    const delegatorKeys = createKeys('delegatorKeys');
+    const insideTx = ExtrinsicHelper.api.tx.msa.createSponsoredAccountWithDelegation(
+      delegatorKeys.publicKey,
+      signPayloadSr25519(delegatorKeys, addProviderData),
+      addProviderPayload
+    );
+    const tx = ExtrinsicHelper.api.tx.frequencyTxPayment.payWithCapacity(insideTx);
+    const feeDetails: FeeDetails = await ExtrinsicHelper.apiPromise.rpc.frequencyTxPayment.computeCapacityFeeDetails(
+      tx.toHex(),
+      null
+    );
+    assert.notEqual(feeDetails, undefined, 'should have returned a feeDetails');
+    assert.notEqual(feeDetails.inclusionFee, undefined, 'should have returned a partialFee');
+    assert(feeDetails.inclusionFee.isSome, 'should have returned a partialFee');
+    const { baseFee, lenFee, adjustedWeightFee } = feeDetails.inclusionFee.toJSON() as any;
+
+    // payWithCapacity costs will fluctuate, so just checking that they are valid positive numbers
+    assert(baseFee > 1, 'The base fee appears to be wrong.');
+    assert(lenFee > 1, 'The len fee appears to be wrong.');
+    assert(adjustedWeightFee > 1, 'The adjusted weight fee appears to be wrong');
+  });
+
+  it('Returns nothing when requesting capacity cost of a non-capacity transaction', async function () {
+    const tx = ExtrinsicHelper.api.tx.msa.retireMsa();
+    const feeDetails: FeeDetails = await ExtrinsicHelper.apiPromise.rpc.frequencyTxPayment.computeCapacityFeeDetails(
+      tx.toHex(),
+      null
+    );
+    assert.notEqual(feeDetails, undefined, 'should have returned a feeDetails');
+    assert(feeDetails.inclusionFee.isNone, 'should have returned something for the inclusionFee');
```

Comment:
Shouldn't this message be
```suggestion
    assert(feeDetails.inclusionFee.isNone, 'should have returned None for the inclusionFee');
```